Xkcd-Comics/ViewModels/ComicsViewModel.swift

```diff
@@ -0,0 +1,138 @@
+    
+    func fetchComic() {
+        fetchingService.fetchComic(comicType: .latestComic) { result in
+            switch result {
+            case .success(let comic):
+                self.handleFetchedComic(comic: comic)
+            case .failure(let error):
+                print(error.localizedDescription)
+            }
+        }
+    }
+    
+    func fetchPreviousComic() {
+        print("VM fetchPreviousComic: number of current comic: \(comic.num)")
+        fetchingService.fetchComic(comicType: .specificComic(comic.num - 1)) { result in
+            switch result {
+            case .success(let comic):
+                self.handleFetchedComic(comic: comic)
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    self.isShowingPreviousComicAlert = true
+                }
+                print("ComicsVM, fetching previous comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func fetchNextComic() {
+        fetchingService.fetchComic(comicType: .specificComic(comic.num + 1)) { result in
+            switch result {
+            case .success(let comic):
+                self.handleFetchedComic(comic: comic)
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    self.isShowingNextComicAlert = true
+                }
+                print("ComicsVM, fetching next comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func toggleDescription() {
+        isShowingExplanation.toggle()
+    }
+    
+    func toggleInfo() {
+        isShowingInfo.toggle()
+    }
+    
+    func searchComic(searchNum: String) {
+        guard isShowingSearch else {
+            isShowingSearch = true
+            return
+        }
+        // TODO: error warning if the input is not int
+        fetchingService.fetchComic(comicType: .specificComic(Int(searchNum) ?? 1)) { result in
+            switch result {
+            case .success(let comic):
+                DispatchQueue.main.async {
+                    self.comic = comic
+                    _ = self.isSaved(comic: comic)
+                    self.isShowingSearch = false
+                }
+            case .failure(let error):
+                print("ComicsVM, searching next comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func saveAsFavourite(comic: Comic) {
+        guard !isSaved(comic: comic) else {
+            print("already saved")
+            return
+        }
+        DispatchQueue.main.async {
+            _ = self.isSaved(comic: comic)
+        }
+        userDataService.addComicToFavorites(comic: comic)
+    }
+    
+    func getFavoriteComics() {
+        guard let comics = userDataService.favoriteComics() else {
+            return
+        }
+        for comic in comics {
+            print("ComicVM, getting favorite comics: \(comic.title)")
+        }
+    }
+    
+    private func isSaved(comic: Comic) -> Bool {
```

Comment:
A method should typically either return something _or_ have side effects - not both. This method both mutates the isSaved state and returns a value, which is confusing. The convention is that if you call a method called `is...`, then you just get a return value and should not have to worry about anything being modified - you're just asking a question and would be surprised that the asking also caused something to be modified. 

By separating the two, you would also reduce the amount of code needed. In this method, instead of having both `self.isSaved = false` and `return false` three times, you could just have `return false`.  And then, the caller of the method would decide what to do with that information. Either store it with `self.isSaved = isSaved(comic: comic)` or just use it in a condition like `guard !isSaved(comic: comic)`.